pkg/watch/replay.go

```diff
@@ -77,21 +80,26 @@ func (wm *Manager) replayEventsLoop() error {
 				continue
 			}
 
+			if req.r.events == nil && req.r.initialPopulation == nil {
```

Comment:
This dual-channel-writing logic is another exceptional case that makes me think we should disentangle the readiness checker and the watch manager